pkg/plugin/processor/builtin/util_test.go

```diff
@@ -0,0 +1,53 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package builtin
+
+import (
+	"testing"
+
+	sdk "github.com/conduitio/conduit-processor-sdk"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/matryer/is"
+)
+
+func AreEqual(t *testing.T, want, got sdk.ProcessedRecord) {
+	is := is.New(t)
+
+	if wantErr, ok := want.(sdk.ErrorRecord); ok {
+		gotErr, gotOk := got.(sdk.ErrorRecord)
+		if gotOk {
+			is.Equal(wantErr.Error.Error(), gotErr.Error.Error())
+		}
+	}
```

Comment:
Do we need this part? seems like the comparer below is taking care of comparing the errors. And if that's so, we might just get away with having a global variable with the options, like:

```go
var cmpProcessedRecordOpts = []cmp.Option{ ... }
```

And we would use it in tests directly:

```go
is.Equal("", cmp.Diff(want, got, cmpProcessedRecordOpts...))
```

My issue with these custom assert functions is that I don't know what they are comparing until I read them. If I see `is.Equal` I know we're asserting the equality, and then I see `cmp.Diff` I know we're diffing two records and we expect no diff. My 2 cents 🙃

Reply:
Agreed, it's not that pretty. If the name is the issue, I'll try suggesting a better one (AreRecordsEqual or smth along those lines).

A few times I needed a simple way to compare processed records. Additionally, I wanted `cmp.Diff` because it shows the difference very nicely.

The main issue in tests with `sdk.ErrorRecord` is that the errors themselves cannot be easily compared (well, unless you use `cerrors.Is/As`, but that's not always nice). That's why I need a separate check for ErrorRecords.

Reply:
The name is not the issue, also `cmp.Diff` is fine (I like it especially for comparing complex objects like records). I'm just suggesting keeping the assertion in the test instead of having a separate assertion function. The functionality would still be the same, it's just the readability that changes.

Regarding errors, I'm trying to say that the code I'm commenting on (the loop containing `is.Equal(wantErr.Error.Error(), gotErr.Error.Error())`) seems redundant, as the comparer passed to `cmp.Diff` will produce the same output, for example:

```go
	unwrap_debezium_test.go:235:  !=   sdk.ErrorRecord{
		- 	Error: e"expected error message",
		+ 	Error: e"data to be unwrapped doesn't contain a payload field",
		  }
```

Here's essentially what I'm suggesting: https://github.com/ConduitIO/conduit/commit/51acade531e149503f35ab52dfa2496e70814fd2.

Reply:
Ah, I understand what you mean now. Looks good!